apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -0,0 +1,201 @@
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxstatements: 200
+ */
+/* global $ */
+'use strict';
+
+var NetSimLogEntry = require('./NetSimLogEntry');
+var Packet = require('./Packet');
+var markup = require('./NetSimRouterLogModal.html.ejs');
+var netsimGlobals = require('./netsimGlobals');
+
+/**
+ * Generator and controller for contents of modal dialog that reveals
+ * all router logs together, in a searchable/sortable/filterable manner.
+ *
+ * @param {jQuery} rootDiv
+ * @constructor
+ */
+var NetSimRouterLogModal = module.exports = function (rootDiv) {
+  /**
+   * Component root, which we fill whenever we call render()
+   * @type {jQuery}
+   * @private
+   */
+  this.rootDiv_ = rootDiv;
+
+  /**
+   * @type {NetSimShard}
+   * @private
+   */
+  this.shard_ = null;
+
+  /**
+   * @type {NetSimLogEntry[]}
+   * @private
+   */
+  this.logEntries_ = [];
+
+  /**
+   * Tracking information for which events we're registered to, so we can
+   * perform cleanup as needed.
+   * @type {Object}
+   * @private
+   */
+  this.eventKeys_ = {};
+
+  /**
+   * Sorting key, changed by user interaction, which determines which sort
+   * we use on render.
+   * @type {string}
+   * @private
+   */
+  this.sortBy_ = 'timestamp';
+
+  /**
+   * Whether currently using a descending sort.
+   * @type {boolean}
+   * @private
+   */
+  this.sortDescending_ = true;
+
+  this.render();
+};
+
+function makeStringSorter(logEntryToSortString) {
+  return function (logEntries) {
+    var sortProxy = logEntries.map(function (entry, index) {
+      return {
+        index: index,
+        sortValue: logEntryToSortString(entry)
+      };
+    });
+
+    sortProxy.sort(function (a, b) {
+      return a.sortValue.localeCompare(b.sortValue);
+    });
+
+    return sortProxy.map(function (proxy) {
+      return logEntries[proxy.index];
+    });
+  };
+}
```

Comment:
Thanks for pointing this out, it would definitely be better - especially because `Array.prototype.sort` is not necessarily stable, and `_.sortBy` is guaranteed to be (friendlier for letting people ad-hoc sort by sender and then time).